### PR TITLE
chore(telemetry): enable telemetry by default

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

Enables telemetry collection when an application is run with `ddtrace-run`.

## Checklist
- [x] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [x] Add additional sections for `feat` and `fix` pull requests.
- [x] Ensure tests are passing for affected code.
- [x] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation

Collect tracer configurations.

## Design 

Starts a telemetry writer which sends a payload of tracer configuration to the datadog agent. As of now one event is sent on application startup and another is sent when an application terminates.

## Testing strategy

Ensure telemetry test suite passes in ci.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: releasenotes/notes/enable-telemetry-by-default-ae39651a478f8e61.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Enable telemetry collection when using ddtrace-run.
```
Comment (mabdinur) [inline releasenotes/notes/enable-telemetry-by-default-ae39651a478f8e61.yaml:4]: TO DO update release note

**Review by ZStriker19 — DISMISSED** [review]

File: releasenotes/notes/enable-telemetry-by-default-ae39651a478f8e61.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Enable :ref:`telemetry <Instrumentation Telemetry>` collection when using ddtrace-run.
```
Comment (brettlangdon) [inline releasenotes/notes/enable-telemetry-by-default-ae39651a478f8e61.yaml:4]:
```suggestion
    Enable :ref:`telemetry <Instrumentation Telemetry>` collection when using ``ddtrace-run``.
```

Does spelling complain about this? not sure this suggestion matters

File: docs/index.rst [hunk 3]
```diff
@@ -183,6 +183,18 @@ contacting support.
...
+
+.. _`Instrumentation Telemetry`:
+
+Instrumentation Telemetry
+-------------------------
+
+Datadog may gather environmental and diagnostic information about your tracing libraries; this includes information
+about the host running an application, operating system, programming language and runtime, APM integrations used,
+and application dependencies. Additionally, Datadog may collect information such as diagnostic logs, crash dumps
+with obfuscated stack traces, and various system performance metrics.
```
Comment (brettlangdon) [inline docs/index.rst:195]: should we add a statement on how to disable?

> To disable set `DD_INSTRUMENTATION_TELEMETRY_ENABLED=false` environment variable

?

File: docs/configuration.rst [hunk 4]
```diff
@@ -64,6 +64,12 @@ below:
      - Enable sending of spans to the Agent. Note that instrumentation will still be installed and spans will be
        generated. Added in ``v0.41.0`` (formerly named ``DATADOG_TRACE_ENABLED``).
 
+       .. _dd-instrumentation-telemetry-enabled:
+   * - ``DD_INSTRUMENTATION_TELEMETRY_ENABLED``
+     - Boolean
+     - True
+     - Enables sending telemetry events to the agent.
```
Comment (brettlangdon) [inline docs/configuration.rst:71]: Should we link to Instrumentation telemetry section here?

**Review by brettlangdon — COMMENTED** [review]
minor nits, but otherwise lgtm

**Review by brettlangdon — DISMISSED** [review]

**Comment by mabdinur** [comment]
@Mergifyio backport 1.4

File: docs/index.rst [hunk 5]
```diff
@@ -183,6 +183,19 @@ contacting support.
 
 .. [5] ``grpc.aio`` is automatically instrumented starting with ``grpcio>=1.32.0``.
 
+
+.. _`Instrumentation Telemetry`:
+
+Instrumentation Telemetry
+-------------------------
+
+Datadog may gather environmental and diagnostic information about your tracing libraries; this includes information
```
Comment (majorgreys) [inline docs/index.rst:192]: Not sure about "your tracing" here.

```suggestion
Datadog may gather environmental and diagnostic information about instrumentation libraries; this includes information
```

**Review by Kyle-Verhoog — COMMENTED** [review]
can we add test cases to ensure that this works, otherwise lgtm!

File: docs/index.rst [hunk 6]
```diff
@@ -185,6 +185,19 @@ contacting support.
...
+Instrumentation Telemetry
+-------------------------
+
+Datadog may gather environmental and diagnostic information about instrumentation libraries; this includes information
+about the host running an application, operating system, programming language and runtime, APM integrations used,
+and application dependencies. Additionally, Datadog may collect information such as diagnostic logs, crash dumps
+with obfuscated stack traces, and various system performance metrics.
+
+To disable set ``DD_INSTRUMENTATION_TELEMETRY_ENABLED=false`` environment variable.
+
```
Comment (mabdinur) [inline docs/index.rst:200]:
```suggestion

See our official :ref`datadog documentation <https://docs.datadoghq.com/tracing/configure_data_security#telemetry-collection>` for more details
```

**Review by Yun-Kim — APPROVED** [review]

**Pull request merged**